boards: fanke: Add support for FK743M5-XIH6

realhonbo opened the pull request:
The following features have been tested to work on the hardware:
- led
- uart1
- spi lcd
- nor flash

realhonbo commented:
Hi @kartben, could you please review this PR when you have time? 🙏

kartben commented on boards/fanke/fk743m5_xih6/doc/img/fk743m5-xih6.webp:
Filename should be same as the board so replace the hyphen by an underscore

kartben commented on boards/fanke/fk743m5_xih6/doc/index.rst:
```diff
@@ -0,0 +1,129 @@
+FK743M5-XIH6 board has 5 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
```
```suggestion
Default Zephyr Peripheral Mapping
---------------------------------
```

realhonbo replied:
thank you very much, I have patched it. @kartben

kartben commented on boards/fanke/fk743m5_xih6/fk743m5_xih6.yaml:
```diff
@@ -0,0 +1,13 @@
+flash: 2048
+supported:
+  - uart
+  - gpio
```
Looks like the list can be expanded (spi, ...)

kartben commented:
@realhonbo please squash your commits so that there is just one (this will require a force push to your branch)

realhonbo commented:
> @realhonbo please squash your commits so that there is just one (this will require a force push to your branch)

Thank you, I have merged them to one

realhonbo commented on boards/fanke/fk743m5_xih6/fk743m5_xih6.dts:
```diff
@@ -0,0 +1,172 @@
+		led0 = &user_led;
+	};
+
+	mipi_dbi_st7789v_320x240 {
```
Your perseverance is inspiring. However, I think since it's a complementary product, perhaps I should provide and then disable it, for the reference of users.

kartben replied:
@realhonbo can you elaborate on why the other PR was closed?

realhonbo replied:
I am not avoid review. In that PR, I thought I should merge patchs to one push, but afraid zephyr do not receive --force

realhonbo replied:
Okay, I remove them. But I see the other fanke and weact board also have the st7789v in dts, maybe they should be changed

realhonbo commented on boards/fanke/fk743m5_xih6/fk743m5_xih6.dts:
```diff
@@ -0,0 +1,120 @@
+#include <st/h7/stm32h743xihx-pinctrl.dtsi>
+#include <zephyr/dt-bindings/memory-attr/memory-attr-arm.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+#include <zephyr/dt-bindings/mipi_dbi/mipi_dbi.h>
```
OKay, I've remove them except`#include <st/h7/stm32h743xihx-pinctrl.dtsi>`, or compile failed

realhonbo commented on boards/fanke/fk743m5_xih6/fk743m5_xih6.dts:
```diff
@@ -0,0 +1,117 @@
+	/* Winbond external flash */
+	w25q64_qspi: qspi-nor-flash@90000000 {
+		compatible = "st,stm32-qspi-nor";
+		reg = <0x90000000 DT_SIZE_M(64)>; /* 64 Mbits */
```
Yeah, you are right. I have corrected it.

JarmouniA reviewed (approved):
Nice work!

realhonbo commented:
> Nice work!

Thanks for your help. There are 2 checks of  "Documentation Build" that failed, which prevented the merge. Do I need to fix them?

realhonbo commented:
> > 我需要修复它们吗？
> 
> 是的，`python ./scripts/ci/check_compliance.py --commits HEAD~1..HEAD`在强制推送之前，先在本地检查合规性。

yeah, I've checked it before. But the failing still exist

Running Identity         tests in /home/x/zephyr_projects/zephyr ...
Running Gitlint          tests in /home/x/zephyr_projects/zephyr ...
Running DevicetreeBindings tests in /home/x/zephyr_projects/zephyr ...
Running Nits             tests in /home/x/zephyr_projects/zephyr ...
Running KconfigBasicNoModules tests in /home/x/zephyr_projects/zephyr ...
Running SysbuildKconfigBasic tests in /home/x/zephyr_projects/zephyr ...
Running Ruff             tests in /home/x/zephyr_projects/zephyr ...
Running ClangFormat      tests in /home/x/zephyr_projects/zephyr ...
Running SphinxLint       tests in /home/x/zephyr_projects/zephyr ...
Running MaintainersFormat tests in /home/x/zephyr_projects/zephyr ...
Running Checkpatch       tests in /home/x/zephyr_projects/zephyr ...
Running BinaryFiles      tests in /home/x/zephyr_projects/zephyr ...
Running Pylint           tests in /home/x/zephyr_projects/zephyr ...
Running KconfigBasic     tests in /home/x/zephyr_projects/zephyr ...
Running GitDiffCheck     tests in /home/x/zephyr_projects/zephyr ...
Running KconfigHWMv2     tests in /home/x/zephyr_projects/zephyr ...
Running SysbuildKconfigBasicNoModules tests in /home/x/zephyr_projects/zephyr ...
Running Kconfig          tests in /home/x/zephyr_projects/zephyr ...
Running SysbuildKconfig  tests in /home/x/zephyr_projects/zephyr ...
Running TextEncoding     tests in /home/x/zephyr_projects/zephyr ...
Running KeepSorted       tests in /home/x/zephyr_projects/zephyr ...
Running ModulesMaintainers tests in /home/x/zephyr_projects/zephyr ...
Running BoardYml         tests in /home/x/zephyr_projects/zephyr ...
Running YAMLLint         tests in /home/x/zephyr_projects/zephyr ...
Running ImageSize        tests in /home/x/zephyr_projects/zephyr ...

Complete results in compliance.xml

kartben commented:
@realhonbo the CI error is not related to your change. I'm looking into it 
https://discord.com/channels/720317445772017664/733037890514321419/1350474875780989031

realhonbo commented:
I retracted the PR I just pushed because it should open a new branch

nordicjm commented on boards/fanke/fk743m5_xih6/doc/img/fk743m5_xih6.webp:
can resolution of image be reduced?

realhonbo replied:
OK, I've resized and zipped it to 34.2KB

realhonbo commented on boards/shields/st7789v_generic/st7789v_fanke.overlay:
What? I'm checking my PR

realhonbo committed:
boards: fanke: Add support for FK743M5-XIH6

Signed-off-by: Honbo He <hehongbo918@gmail.com>

The following features have been tested to work on the hardware:
- led
- uart1
- spi lcd
- nor flash

realhonbo commented:
I've picked it and ensure only 9 files in dir "boards/fanke/fk743m5_xih6".
Sorry about that, branchs in my pc had misled me a little

kartben commented on boards/fanke/fk743m5_xih6/board.cmake:
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(openocd --target-handle=_CHIPNAME.cpu0)
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
```
Wonder if it should be made the default runner?
